lib/record_store/zone.rb

```diff
@@ -56,22 +56,49 @@ def yaml_files
     validate :validate_provider_can_handle_zone_records
 
     def self.from_yaml_definition(name, definition)
```

Comment:
Given it can be a single or multiple YAML definitions, it may make sense to rename this function to `load`.